src/frontend/ts/state/chessStore.ts

```diff
@@ -132,11 +120,6 @@ const useChessStore = create<ChessState>((set, get) => ({
         const resultData = data as ResultUpdate
         const gameID = resultData.gameID
 
-        const displayGame = get().displayGame
-        if ((displayGame !== null) && (gameID === displayGame.info.gameID)) {
-          get().setDisplayGame(null)
-        }
-
```

Comment:
I think we need to make 3 changes for correct behaviour:
1. Reorder all of the cards in `app/chess.hoon`, so that we send the `ArchivedGameInfo` first, *then* we send the Result, which will:
    1. Remove the game from `activeGames`
    2. Compare the `gameID` with `displayGame.info.gameID`
    3. If they match, it will set `displayGame` to `archivedGames.get(gameID)`
2. Add a new mark which sends an `ArchivedGameInfo` *with moves*. This mark will be used whenever a game is completed, and the current mark will only be used for sending the initial archive (i.e. on line `702` of `app/chess.hoon`, inside `+on-watch`)
3. `fetchArchivedMoves` should only call `scryMoves` if `currentGame.moves` is `null` or has length `0`

Reply:
wow, the order of the cards matter!?
I always considered it like a %= in a trap, where all effects/changes occur at the same time. but it makes sense that when you send an effect out it necessarily has an order.

is the order top to bottom, first to last?

Reply:
Actually, nevermind: an even better solution exists! Keep the marks as is, and send the archived game without `moves`. When we get a `Result` update in the frontend, we should copy the moves from the `activeGame` into the `archivedGame` before deleting the `activeGame`!

There might be a bug here where we'll be missing the last move in the archive, but that's easy to spot & fix.

Reply:
@brbenji That's weird - missed your post because I tried to make one at the same time. That's right, the card order matters: it's the order in which they're found in the `list` you pass back to Gall (first to last). There's also discussion over [what order should be used to process cards in the kernel](https://github.com/urbit/urbit/pull/6041).

Reply:
neat!

Reply:
> Actually, nevermind regarding (2): an even better solution exists! Keep the marks as is, and send the archived game without `moves`. When we get a `Result` update in the frontend, we should copy the moves from the `activeGame` into the `archivedGame` before deleting the `activeGame`!
> 
> There might be a bug here where we'll be missing the last move in the archive, but that's easy to spot & fix.

@ashelkovnykov Oh...I just finished updating the PR with your changes. I'm running low on time for today and this weekend.

things seem to be working on mars, but the %chess app opens to a blank page complaining that the `displayGame` is `null`. I'm not sure where we establish the practiceBoard is the first thing in `displayGame`?

Reply:
feel free to take this to the finish line over the weekend. I'm out of time, but can come back to it monday.